crates/rpc/rpc-types-compat/src/transaction/mod.rs

```diff
@@ -102,7 +102,9 @@ impl TransactionCompat for () {
         WithOtherFields::default()
     }
 
-    fn otterscan_api_truncate_input(_tx: &mut Self::Transaction) {}
+    fn otterscan_api_truncate_input(tx: &mut Self::Transaction) {
+        tx.input = tx.input.slice(..4);
```

Comment:
this is actually dangerous and incorrect in the other implementations as well, we need to check that the input slice actually as at least 4 items

Reply:
it is [double-check](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/otterscan.rs#L232) in case of

Reply:
I see, but this check should be in the impl instead, can do that separately, though